core/object/class_db.cpp

```diff
@@ -1678,6 +1678,13 @@ Variant ClassDB::class_get_default_property_value(const StringName &p_class, con
 				if (E.usage & (PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR)) {
 					if (!default_values[p_class].has(E.name)) {
 						Variant v = c->get(E.name);
+						if (v.get_type() == Variant::OBJECT) {
+							Object* o = v.get_validated_object();
+							if(o) {
+								// Skip non-null objects, caching a specific instance causes unexpected behavior.
+								continue;
+							}
+						}
```

Comment:
```suggestion
						if (v.get_type() == Variant::OBJECT && v.get_validated_object() != nullptr) {
							// Skip non-null objects, caching a specific instance causes unexpected behavior.
							continue;
						}
```
No need to grab the value itself if not used, IMO